### PR TITLE
[AGENT-11268] Add a performance tuning note about increasing records per poll

### PR DESCRIPTION
**Pull request by jszwedko**
Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,9 +111,16 @@ A REST call can be executed against one of the cluster instances, and the config
 
 To improve performance of the connector, you can try the following options:
 
-* Increase the number of parallel tasks by adjusting the `tasks.max` parameter. Only do this if the hardware is 
-underutilized, such as low CPU, low memory usage and low data injection throughput. Do not set more tasks than partitions.
-* Increase hardware resources on cluster nodes in case of resource exhaustion, such as high CPU, or high memory usage.
+* Update the number of records fetch per poll by setting
```
Comment (lukesteensen) [inline README.md:114]:
```suggestion
* Update the number of records fetched per poll by setting
```

**Review by tobz — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -111,9 +111,16 @@ A REST call can be executed against one of the cluster instances, and the config
 
 To improve performance of the connector, you can try the following options:
 
-* Increase the number of parallel tasks by adjusting the `tasks.max` parameter. Only do this if the hardware is 
-underutilized, such as low CPU, low memory usage and low data injection throughput. Do not set more tasks than partitions.
-* Increase hardware resources on cluster nodes in case of resource exhaustion, such as high CPU, or high memory usage.
+* Update the number of records fetched per poll by setting
+  `consumer.override.max.poll.records` in the plugin configuration. This plugin
+  sends batches of records synchronously with each poll so a low number of records
+  per poll will reduce throughput. Consider setting this to 500 or 1000.
```
Comment (estherk15) [inline README.md:117]: Non-blocking, why are sentences broken up like this? Is it a formatting thing?

File: README.md [hunk 3]
```diff
@@ -111,9 +111,16 @@ A REST call can be executed against one of the cluster instances, and the config
...
+* Increase the number of parallel tasks by adjusting the `tasks.max` parameter.
+  Only do this if the hardware is underutilized, such as low CPU, low memory
+  usage and low data injection throughput. Do not set more tasks than
```
Comment (estherk15) [inline README.md:120]:
```suggestion
  usage, and low data injection throughput. Do not set more tasks than
```

**Review by estherk15 — APPROVED** [review]
Added an oxford comma.

File: README.md (hunk 2 above)
Comment (jszwedko) [reply README.md:117]: Ah, yeah, this was just word wrapping. When it is rendered as markdown, it'll all be one line that overflows.

**Pull request merged**